lua/wire/wireshared.lua

```diff
@@ -369,6 +369,45 @@ function WireLib.ErrorNoHalt(message)
 	end
 end
 
+--- Generate a random version 4 UUID and return it as a string.
+function WireLib.GenerateUUID()
+	-- It would be easier to generate this by word rather than by byte, but
+	-- MSVC's RAND_MAX = 0x7FFF, which means math.random(0, 0xFFFF) won't
+	-- return all possible values.
+	local bytes = {}
+	for i = 1, 16 do bytes[i] = math.random(0, 0xFF) end
+	bytes[7] = bit.bor(0x40, bit.band(bytes[7], 0x0F))
+	bytes[9] = bit.bor(0x80, bit.band(bytes[7], 0x3F))
+	return string.format("%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x", unpack(bytes))
+end
+
+local UUID_MESSAGE_NAME = "WireLib.GetServerUUID"
+if SERVER then
+	--- Return a persistent UUID associated with the server.
+	function WireLib.GetServerUUID()
+		local uuid = cookie.GetString(UUID_MESSAGE_NAME)
+		if not uuid then
+			uuid = WireLib.GenerateUUID()
+			cookie.Set(UUID_MESSAGE_NAME, uuid)
+		end
+		return uuid
+	end
+
+	util.AddNetworkString(UUID_MESSAGE_NAME)
+
+	hook.Add("PlayerInitialSpawn", UUID_MESSAGE_NAME, function(player)
+		net.Start(UUID_MESSAGE_NAME)
```

Comment:
Oh never mind the previous comment...
But should we really send stuff we don't need (yet)?
Also, I think @Divran told me umsg is better for tiny messages like this.